extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -587,16 +618,15 @@ export class AmpAnalytics extends AMP.BaseElement {
     if (!spec) {
       return resolve;
     }
-    if (!spec['sampleOn']) {
+    const sampleOn = spec['sampleOn'];
+    if (!sampleOn) {
       user().error(TAG, 'Invalid sampleOn value.');
       return resolve;
     }
     const threshold = parseFloat(spec['threshold']); // Threshold can be NaN.
     if (threshold >= 0 && threshold <= 100) {
-      const keyPromise = this.variableService_.expandTemplate(
-          spec['sampleOn'], this.expansionOptions_({}, trigger))
-        .then(key => urlReplacementsForDoc(this.element).expandAsync(key));
-      return keyPromise
+      const expansionOptions = this.expansionOptions_({}, trigger);
+      return this.expandTemplateWithUrlParams_(sampleOn, expansionOptions)
           .then(key => this.cryptoService_.uniform(key))
           .then(digest => digest * 100 < spec['threshold']);
```

Comment:
Could you please change here `spec['threshold']` to `threshold` from above `const threshold`?